extension/intents/nicknames/routineExecutor.js

```diff
@@ -0,0 +1,48 @@
+/* globals log */
+
+import * as intentRunner from "../../background/intentRunner.js";
+
+export class RoutineExecutor {
+  constructor(routineName, subcommands, index = 0) {
+    this.routineName = routineName;
+    this.subcommands = subcommands;
+    this.index = index;
+  }
+
+  async runSubcommand(subcommand) {
+    let hadError = false;
+    subcommand.onError = () => {
+      hadError = true;
+    };
+
+    subcommand.parentRoutine = {
+      name: this.routineName,
+      nextIndex: this.index + 1,
+      stop: false,
+    };
```

Comment:
I think instead I would do something like `subcommand.routineExecutor = this` and then pause can access methods on this object.